controllers/vsl.go

```diff
@@ -1,8 +1,9 @@
 package controllers
```

Comment:
.go files shouldn't be part of this PR?

Reply:
@kaovilai my mistake. I was working on two things and forgot to switch branches. Good catch